test/integration/auth/scram_sha_256.test.ts

```diff
@@ -0,0 +1,41 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { deps, type MongoClient } from '../../mongodb';
+
+describe('SCRAM_SHA_256', function () {
+  context('when saslprep is not a function', () => {
```

Comment:
Should we add the happy path, just a expect(saslprep).to.have.been.called when it is a function

Reply:
Since we have no SCRAM tests, I chose to only write regression tests.  What value do you think that test provides, in the context of this bug fix?

Reply:
its testing the inverse of the condition, real scram tests would cover that saslprep is being used / works as expected, I'm just looking to cover the code change here.